### PR TITLE
Feature/#122 file download

### LINKED ISSUE #122: As a lecturer or a student, i want to be able to download stored files.
Dep: #83 

AC:
- [ ] Clicking a button / following a URL, will start a download of any uploaded file specified (--> #83)

_Hint:_ From this point, we're not thinking about restricting access to files. That may come later, in a separate story.

### PR DESCRIPTION
**Pull request by caustt**
# Resolves #122

This PR adds / introduces the feature of downloading an uploaded file 

Acceptance Criteria:
- [x] Clicking a button / following a URL, will start a download of any uploaded file specified 

Testing steps:
- go to /uploaded_files
- upload a file
- there should be a download link now at /uploaded_files
- clicking that link or following URL /uploaded_files/1 should download that file

Steps needed for migration:
- None

Definition Of Done:
- [x] [Class-Diagram](https://www.lucidchart.com/invitations/accept/705a122b-58a9-4dfa-8b16-c936bc0a46bc) updated

### CODE REVIEW

**Comment by adrianjost** [comment]
- [ ] is blocked by #116

**Comment by WorldofJARcraft** [comment]
The migrations of this branch did not work on my machine. When db/migrate/20191219102226_add_files_schema.rb is run after the migrations from dev (with merged #83 ), it fails because uploaded_files already exists.

File: spec/factories/uploaded_files.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :uploaded_file do
+    content_type { "MyString" }
+    filename { "MyString" }
+    data { "" }
+  end
+end
```
Comment (WorldofJARcraft) [inline spec/factories/uploaded_files.rb]: Never used.

**Comment by adrianjost** [comment]
@WorldofJARcraft All your requested changes where regarding code that was already merged into dev. They where introduced by your upload feature branch #116

**Pull request merged**